blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/MultibranchPipelineRunContainer.java

```diff
@@ -85,22 +83,26 @@ public BlueRun get(String name) {
             sortBranchesByLatestRun(branches);
         }
 
+        List<BlueRun> allRunList = new ArrayList<BlueRun>();
         for (final BluePipeline b : branches) {
             BlueRunContainer blueRunContainer = b.getRuns();
             if(blueRunContainer==null){
                 continue;
             }
-            Iterator<BlueRun> it = blueRunContainer.iterator(0, MAX_MBP_RUNS_ROWS);
-            int count = 0;
-            Utils.skip(it, start);
-            while (it.hasNext() && count++ < limit) {
-                c.add(it.next());
+            List<BlueRun> it = Lists.newArrayList(blueRunContainer.list());
+            allRunList.addAll(it);
+            if (allRunList.size() > start + limit){
+                break;
             }
         }
 
-        Collections.sort(c, LATEST_RUN_START_TIME_COMPARATOR);
+        if (start < allRunList.size()) {
```

Comment:
I'm trying to follow this, so i'm writing it out.

If start (3) is smaller than the total size (5), then grab start (3) to the end(5)
so grabs 2 element?
Then sorts that list of 2 elements
then limits it (1)
So ends up with 1 correct?

That feels right to me, though seems wasteful to grab all runs to do it, but looking at the old code it did the same thing.